packages/components/src/components/modal/modal.tsx

```diff
@@ -194,7 +194,7 @@ const ModalElement = ({
                     )}
                     {has_close_icon && (
                         <div onClick={toggleModal} className='dc-modal-header__close' role='button'>
-                            <Icon icon='IcCross' color={close_icon_color} />
+                            <Icon icon='IcCross' color={close_icon_color} data_testid='dt-close-modal-icon' />
```

Comment:
```suggestion
                            <Icon icon='IcCross' color={close_icon_color} data_testid='dt_close_modal_icon' />
```

Reply:
Cant we just mock Icon component in the test itself rather than doing this ? 🤔

Reply:
hmmm we could but then wouldn't this be allowing the test cases to have less lines of code, since we just need to check for the icon's test id rather than mocking it entirely, and then in other components we don't need to keep on mocking it